pkg/util/steps/refreshing.go

```diff
@@ -70,10 +70,20 @@ func (s authorizationRefreshingActionStep) run(ctx context.Context, log *logrus.
 			(azureerrors.HasAuthorizationFailedError(err) ||
 				azureerrors.HasLinkedAuthorizationFailedError(err)) {
 			log.Print(err)
-			// https://github.com/Azure/ARO-RP/issues/541: it is unclear if this
-			// refresh helps or not
-			err = s.authorizer.RefreshWithContext(ctx)
-			return false, err
+			// Try refreshing auth. If auth refresh error occurs, return
+			// that instead of original auth error.
+			err = wait.PollImmediateUntil(pollInterval, func() (bool, error) {
```

Comment:
this also doesn't look right.  In validateActions() above we avoid having nested wait.Polls; I think we can use exactly the same trick here to simplify this.

Also, I think the `timeoutCtx.Err() == nil` check on line 69 is a red herring and can be removed.

Reply:
I will simplify it as in `validateActions`, but I don't think that `timeoutCtx.Err() == nil` is unnecessary - it ensures that, after the last attempt to run the step, we do not overwrite the error returned from that step with a timeout error.